master/buildbot/steps/source/git.py

```diff
@@ -770,3 +661,87 @@ def _doTag(self):
 
         ret = yield self._dovccmd(cmd)
         return ret
+
+
+class GitCommit(buildstep.BuildStep, GitStepMixin, CompositeStepMixin):
+
+    description = None
+    descriptionDone = None
+    descriptionSuffix = None
+
+    name = 'gitcommit'
+    renderables = ['files', 'messages']
+
+    def __init__(self, workdir=None, files=None, messages=None, env=None,
```

Comment:
Are directories supported? I think better naming would be `paths`.

Reply:
Yes, paths is a better naming.